lib/middleware/request-handler.ts

```diff
@@ -390,92 +529,36 @@ export function onPollSubscriptions(req: Interface.IOurRequest,
-            return next(new restify.InvalidArgumentError('Invalid unsubscribe data.'));
-        }
-        // Next, validate all correlation Ids
-        // Will error if any invalid correlation Id received
-        var isAllValid = true;
-        _.uniq(req.body.correlationIds).forEach((cid: number): boolean => {
-            if (req.apiSession.activeSubscriptions.has(cid)) {
-                subscriptions.push(req.apiSession.activeSubscriptions.get(cid));
-            } else {
-                isAllValid = false;
-                req.log.debug('Invalid correlation Id ' + cid + ' received.');
-                return false;
-            }
-        });
-        if (!isAllValid) {
-            return next(new restify.InvalidArgumentError('Invalid correlation id.'));
-        }
-    }
-
-    try {
-        var result: Object[] = [];
-        req.blpSession.unsubscribe(subscriptions);
-        subscriptions.forEach((sub: Subscription): void => {
-            sub.removeAllListeners();
-            if (!sub.buffer.isEmpty()) {
-                result.push(sub.buffer.startNewBuffer());
-            }
-            req.apiSession.activeSubscriptions.delete(sub.correlationId);
-            req.apiSession.receivedSubscriptions.delete(sub.correlationId);
-        });
-        req.log.debug({ activeSubscriptions:
-                       req.apiSession.activeSubscriptions.size },
-                      'Unsubscribed.');
+var SUBSCRIPTION_ACTION_MAP: ActionMap = {
+    'start': onSubscribe,
+    'stop':  onUnsubscribe
+};
 
-        // Reset poll Id to null if all subscriptions get unsubscribed
-        if (!req.apiSession.receivedSubscriptions.size) {
-            req.apiSession.lastPollId = req.apiSession.lastSuccessPollId = null;
-        }
-
-        res.sendChunk(result);
-        res.sendEnd(0, 'Unsubscribe Successfully');
-        return next();
-    } catch (err) {
-        req.log.error(err, 'Unsubscription error');
-        return next(new restify.InternalError(err.message));
-    }
+export function getSubscriptionActions(): string[]
+{
+    return _.keys(SUBSCRIPTION_ACTION_MAP);
 }
 
 export function onChangeSubscriptions(req: Interface.IOurRequest,
                                       res: Interface.IOurResponse,
                                       next: restify.Next): void
 {
-    assert(req.apiSession, 'apiSession not found');
-    assert(req.blpSession, 'blpSession not found');
+    dispatchAction(SUBSCRIPTION_ACTION_MAP, req, res, next);
+}
 
-    switch (req.query.action) {
-        case 'start': {
-            return onSubscribe(req, res, next);
-        } break;
-        case 'stop': {
-            return onUnsubscribe(req, res, next);
-        } break;
-        default: {
-            var errorString = 'Invalid subscription action: ' + req.query.action;
-            req.log.error(errorString);
-            return next(new restify.BadRequestError(errorString));
-        } break;
-    }
+var AUTH_ACTION_MAP: ActionMap = {
```

Comment:
Let's put global constants at the top of the file between imports and functions.